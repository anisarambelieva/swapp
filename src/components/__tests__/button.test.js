import { render } from '@testing-library/react';

import Button from '../button.js';

describe('<StyledButton />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button>Click me</Button>);

    expect(container).toMatchSnapshot();
  });

  it('should handle style props', () => {
    const { getByText } = render(
      <Button $color="black" $backgroundColor="white">
        Click me
      </Button>,
    );

    expect(getByText('Click me')).toHaveStyle(
      'color: black; backgroundColor: white;',
    );
  });
});
