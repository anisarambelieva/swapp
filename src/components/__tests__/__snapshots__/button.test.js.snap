// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> should match snapshot 1`] = `
.c0 {
  font-weight: bold;
  border: 0;
}

<div>
  <button
    class="c0 btn btn-primary"
    type="button"
  >
    Click me
  </button>
</div>
`;
